.github/workflows/docs.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e ".[dev, docs]"
+          sudo apt-get install graphviz
+      - name: Compile docs
+        run: |
+          tox -e docs
+          touch build/sphinx/html/.nojekyll  # allow underscores in URL path
+      - name: Deploy docs
+        uses: JamesIves/github-pages-deploy-action@4.1.4
+        with:
+          branch: gh-pages
+          folder: build/sphinx/html
```

Comment:
Let's move this job to `release.yml`. This way
- publishing docs will be part of releasing a new version and we will have a single workflow for this purpose
- publishing docs will happen only when a new version is released instead of each push to the `main` branch, so that the latest pip installable verison and the online docs are synchronous.

Also please rename the job from `docs` to `publish_docs`, the step `Compile docs` to `Build docs`, and the step `Deploy docs` to `Publish to gh-pages`.